pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,362 @@
+        `regularization` must match the length the signal to be inverted.
+
+        Parameters
+        ----------
+        regularization : Signal
+            Regularization as pyfar Signal.
+        """
+        if not isinstance(regularization, pf.Signal):
+            raise ValueError(
+                "Regularization must be a pyfar.Signal object.")
+
+        instance = cls(regu_type = "magnitude spectrum")
+        instance._regu = regularization
+
+        return instance
+
+    def invert(self, signal, beta=1, target=None,
+               normalize_regularization=None):
+        r"""
+        Invert the input using frequency-dependent regularized inversion.
+
+        Parameters
+        ----------
+        signal : Signal
+            Signal to be inverted.
+        beta : float, optional
+            Beta parameter to control the amount of regularization. The default
+            is ``1``.
+        target : Signal, optional
+            Target function for the regularization. The default ``None`` uses
+            constant spectrum with an amplitude of 1 as target.
+        normalize_regularization : str
+            Method of normalization.
+
+            ``'max'``
+                Normalize the regularization :math:`\epsilon(f)` to the maximum
+                magnitude of a given signal :math:`S(f)`.
+
+                :math:`Normalization Factor = \frac{\max(|S(f)|)}{\max(|\epsilon(f)|)}`
+            ``'mean'``
+                Normalize the regularization :math:`\epsilon(f)` to the mean
+                magnitude of a given signal :math:`S(f)`.
+
+                :math:`Normalization Factor = \frac{\text{mean}(|S(f)|)}{\text{mean}(|\epsilon(f)|)}`
+
+            The default ``None`` does not apply any normalization.
+
+
+        Returns
+        -------
+        inverse : Signal
+            The inverted input signal.
+        """ # noqa: E501
+        if target is not None and not isinstance(target, pf.Signal):
+            raise ValueError(
+                "Target function must be a pyfar.Signal object.")
+
+        # zero pad, if target function is given and has different n_samples
+        if target is not None and \
+            target.n_samples != signal.n_samples:
+            n_samples = max(signal.n_samples, target.n_samples)
+            signal = pf.dsp.pad_zeros(signal, (n_samples-signal.n_samples))
+            target = pf.dsp.pad_zeros(target, (n_samples-target.n_samples))
+
+        # get regularization factors
+        regu = self.regularization(signal)
+
+        # Apply optional normalization
+        if normalize_regularization is not None:
+            normalization_factor_ = \
+                self.normalization_factor(signal, normalize_regularization)
+            regu.freq *= normalization_factor_
+
+        data = signal.freq
+
+        # calculate inverse filter
+        inverse = signal.copy()
+        inverse.freq = \
+            np.conj(data) / (np.conj(data) * data + beta *
```

Comment:
I think this is a little faster...
```suggestion
            np.conj(data) / (np.abs(data)**2 + beta *
```